accounts-db/src/accounts_index.rs

```diff
@@ -227,21 +227,50 @@ enum ScanTypes<R: RangeBounds<Pubkey>> {
     Indexed(IndexKey),
 }
 
-/// specification of how much memory in-mem portion of account index can use
+/// specification of threshold limit
 #[derive(Debug, Copy, Clone)]
-pub enum IndexLimitMb {
+pub enum ThresholdLimit {
+    /// threshold based on number of entries
+    Entries(usize),
+    /// threshold based on memory usage in MB
+    MemoryMB(usize),
+}
+
+impl ThresholdLimit {
+    /// Convert threshold limit to entry count
+    ///
+    /// # Arguments
+    /// * `size_of_uninitialized` - bytes for uninitialized entry (Pubkey + Box<AccountMapEntry<T>>)
+    /// * `size_of_single_entry` - bytes for AccountMapEntry<T> with single slot list entry
+    pub fn to_entries(self, size_of_uninitialized: usize, size_of_single_entry: usize) -> usize {
+        match self {
+            ThresholdLimit::Entries(entries) => entries,
+            ThresholdLimit::MemoryMB(mb) => {
+                let bytes_per_entry = size_of_uninitialized + size_of_single_entry;
+                let limit_bytes = mb * 1024 * 1024;
+                limit_bytes / bytes_per_entry
+            }
+        }
+    }
+}
+
+/// specification of how many entries the in-mem portion of account index can hold
+#[derive(Debug, Copy, Clone)]
+pub enum IndexLimit {
     /// use disk index while keeping a minimal amount in-mem
     Minimal,
```

Comment:
I think it could be nice to remove Minimal entirely, and use Threshold(Memory(0)). I think that should mimic today's behavior in master.

Reply:
let's do it in a future Pr.